corehq/apps/users/models.py

```diff
@@ -1627,6 +1627,19 @@ def set_has_built_app(self):
             self.has_built_app = True
             self.save()
 
+    def log_user_create(self, created_by, created_via):
+        # fallback to self if not created by any user
+        created_by = created_by or self
+        if isinstance(created_by, CouchUser):
+            created_by = created_by.get_django_user()
+        log_model_change(
+            created_by,
+            self.get_django_user(use_primary_db=True),
+            message={'created_via': created_via},
+            fields_changed=None,
+            is_create=True
+        )
+
```

Comment:
Avoid calling `get_django_user` twice if `created_by is None`. Also call it with `use_primary_db=True`.
```suggestion
        django_user = self.get_django_user(use_primary_db=True)
        created_by = created_by or django_user
        if isinstance(created_by, CouchUser):
            created_by = created_by.get_django_user()
        log_model_change(
            created_by,
            django_user,
            message={'created_via': created_via},
            fields_changed=None,
            is_create=True
        )

```

Reply:
Good point!

Reply:
https://github.com/dimagi/commcare-hq/pull/27755/commits/0c6a41aac035a8258062f911e1badb851558cd6d

used a different naming